types/builder.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+package types
+
+import (
+	"errors"
+	"fmt"
+)
+
+const (
+	RESERVED_COLUMN_TSID      = "tsid"
+	RESERVED_COLUMN_TIMESTAMP = "timestamp"
+)
+
+type RowBuilder struct {
+	row   *Row
+	built bool
+}
+
+func NewRowBuilder(metric string) *RowBuilder {
+	return &RowBuilder{
+		row: &Row{
+			Series: Series{
+				Metric: metric,
+				Tags:   make(map[string]string),
+			},
+			Fields: make(map[string]interface{}),
+		},
+		built: false,
+	}
+}
+
+func (b *RowBuilder) Reset() *RowBuilder {
```

Comment:
Add comment explain which fields are reserved

Reply:
This reset is the same with `New`, I don't see why this reset is required.

Reply:
The row can then be built again with the same metric